client/lib/path-to-section/index.js

```diff
@@ -17,15 +17,19 @@ const sections = config( 'project' ) === 'wordpress-com' ? require( 'wordpress-c
  *
  *  f( '/' ) === 'reader"
  *  f( '/me' ) === 'me"
- *  f( '/me/account' ) === 'me"
+ *  f( '/me/account' ) === 'account"
  *  f( '/read' ) === 'reader'
  */
 export const wpcomImplementation = path => {
-	const match = find( sections, section =>
-		section.paths.some( sectionPath => startsWith( path, sectionPath ) )
+	// rank matches by the number of characters that match so e.g. /media won't map to /me
+	const rankedMatches = sortBy( sections, section =>
```

Comment:
I agree with @jsnajdr's point here, we don't actually need to sort the array if only looking for the largest element.

We could have something like:
```js
const matches = _.flatMap(sections, sections => 'paths' ).filter( sectionPath => startsWith(sectionPath, path );
const longestMatch = _.maxBy( matches, _.size );

return longestMatch;
```